pkg/util/timeutil/time_zone_util.go

```diff
@@ -66,3 +67,45 @@ func ParseFixedOffsetTimeZone(location string) (offset int, origRepr string, suc
 	}
 	return offset, strings.TrimSuffix(strings.TrimPrefix(origRepr, "("), ")"), true
 }
+
+// TimeZoneOffsetStringConversion converts a time string to offset seconds
+// Supported time zone strings :- GMT/UTC±[00:00:00 - 15:59:59], GMT/UTC±[0-16]
+// Unsupported time zone strings :- GMT/UTC±16:00 to upper, GMT/UTC±6.5
+// (case insensitive)
+func TimeZoneOffsetStringConversion(s string) (offset int64, ok bool) {
+	pattern := `(?i)(GMT|UTC)[+-]((((0?\d)|(1[0-5])):([0-5]\d))|(((0?\d)|(1[0-5])):([0-5]\d):([0-5]\d))|((1+[0-6])|(0?\d)))\b$`
+	re := regexp.MustCompile(pattern)
+	if !(re.MatchString(s)) {
+		return 0, false
+	}
+
+	prefix := "+"
```

Comment:
this one is totally optional: if we use capture groups (https://stackoverflow.com/questions/30483652/how-to-get-capturing-group-functionality-in-go-regular-expressions), i think this can be a one liner

this may also simplify the `split` bit below to not depend on `prefix`

e.g. for `(GMT|UTC)([+-)]`, if we use `re.FindStringSubmatch(s)[0][2]` to get `+` or `-`